pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/mapper/HadoopSegmentCreationMapReduceJob.java

```diff
@@ -205,8 +213,32 @@ protected void map(LongWritable key, Text value, Context context) throws IOExcep
     }
 
     protected void setSegmentNameGenerator(SegmentGeneratorConfig segmentGeneratorConfig, Integer seqId,
-        Path hdfsAvroPath, File dataPath) {
+                                           Path hdfsAvroPath, File dataPath) {
+      Schema schema = segmentGeneratorConfig.getSchema();
+
+      if(schema == null ||
+        schema.getTimeFieldSpec() == null ||
+        schema.getTimeFieldSpec().getIncomingGranularitySpec() == null){
```

Comment:
Can we also add a info log that states schema is null or no time field spec, not setting time column or something aong those lines?